Libs/MRML/Widgets/qMRMLLayoutManager.cxx

```diff
@@ -1330,3 +1330,81 @@ QWidget* qMRMLLayoutManager::viewWidget(vtkMRMLNode* viewNode) const
 
   return d->viewWidget(viewNode);
 }
+
+//------------------------------------------------------------------------------
+void qMRMLLayoutManager::pauseRender(bool pause)
+{
+  // Note: views that are instantiated between pauseRender() and resumeRender() calls will
+  // not be affected by the specified pause state
+
+  Q_D(qMRMLLayoutManager);
+
+  qMRMLLayoutManagerPrivate::PausedViewVector currentPausedViews;
+
+  QString sliceNodeType = "vtkMRMLSliceNode";
+  qMRMLLayoutViewFactory* sliceViewFactory = this->mrmlViewFactory(sliceNodeType);
+  foreach(const QString& viewName, sliceViewFactory->viewNodeNames())
+    {
+    ctkVTKAbstractView* view = (ctkVTKAbstractView*)this->sliceWidget(viewName)->sliceView();
+    qMRMLLayoutManagerPrivate::PausedView pausedView;
+    pausedView.Name = viewName.toStdString();
+    pausedView.NodeType = sliceNodeType.toStdString();
+    pausedView.Paused = view->pauseRender(pause);
+    currentPausedViews.push_back(pausedView);
+    }
+
+  QString threeDNodeType = "vtkMRMLViewNode";
+  qMRMLLayoutViewFactory* threeDViewFactory = this->mrmlViewFactory(threeDNodeType);
+  foreach(const QString& viewName, threeDViewFactory->viewNodeNames())
+    {
+    ctkVTKAbstractView* view = this->threeDWidget(viewName)->threeDView();
+    qMRMLLayoutManagerPrivate::PausedView pausedView;
+    pausedView.Name = viewName.toStdString();
+    pausedView.NodeType = threeDNodeType.toStdString();
+    pausedView.Paused = view->pauseRender(pause);
+    currentPausedViews.push_back(pausedView);
+    }
+  d->PausedViews.push_back(currentPausedViews);
+}
+
+//------------------------------------------------------------------------------
+void qMRMLLayoutManager::resumeRender()
+{
+  // Note: views that are instantiated between pauseRender() and resumeRender() calls will
+  // not be affected by resumeRender()
+
+  Q_D(qMRMLLayoutManager);
+
+  if (d->PausedViews.empty())
+    {
+    qCritical() << "No stored render pause state!";
+    return;
+    }
+
+  QString sliceNodeType = "vtkMRMLSliceNode";
+  QString threeDNodeType = "vtkMRMLViewNode";
+
+  std::vector<qMRMLLayoutManagerPrivate::PausedView> pausedViews = d->PausedViews.back();
```

Comment:
In Qt classes use Qt containers (QVector instead of std::vector) unless there is a specific reason to use std containers.